sa/model.go

```diff
@@ -59,6 +62,50 @@ func badJSONError(msg string, jsonData []byte, err error) error {
 
 const regFields = "id, jwk, jwk_sha256, contact, agreement, initialIP, createdAt, LockCol, status"
 
+// ClearEmail removes the provided email address from one specified registration. If
+// there are multiple email addresses present, it does not modify other ones. If the email
+// address is not present, it does not modify the registration and will return a nil error.
+func ClearEmail(dbMap db.DatabaseMap, ctx context.Context, regID int64, email string) error {
+	_, overallError := db.WithTransaction(ctx, dbMap, func(txWithCtx db.Executor) (interface{}, error) {
+		curr, err := selectRegistration(txWithCtx, "id", regID)
+		if err != nil {
+			return nil, err
+		}
+
+		currPb, err := registrationModelToPb(curr)
+		if err != nil {
+			return nil, err
+		}
+
+		// newContacts will be a copy of currPb.Contact with the email address removed.
+		var newContacts []string
+		for _, contact := range currPb.Contact {
+			if contact == "mailto:"+email {
+				fmt.Fprintf(os.Stderr, "Skipping %s\n", contact)
+			} else {
+				newContacts = append(newContacts, contact)
+			}
+		}
+
+		if slices.Equal(currPb.Contact, newContacts) {
```

Comment:
If you replace the debugging print above with setting a boolean, then this could just be `if !emailCleared {`

Reply:
An earlier version of this change had a boolean `modified`. However, I wrote the logic in the wrong direction the first time, and then Phil had the exact same confusion in review. My conclusion was that explicitly checking for `slices.Equal` after doing the modification is easier to read and more obvious.